### PR TITLE
Add error payloads to create mto shipment in prime api

### PR DESCRIPTION
**Pull request by LeDeep**
## Description

Formats errors for CreateMTOShipment to be consistent and return an error response with the title, detail, and instance ID.

## Reviewer Notes

First time working on these updated error payloads. Are there places that need changes, that I missed?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

Sample request payload
```sh
{
  "body": {
    "moveTaskOrderID": "5d4b25bb-eb04-4c03-0000-ee0398cb779e",
    "shipmentType": "HHG",
    "requestedPickupDate": "2020-06-23",
    "pickupAddress": {
      "id": "c56a4180-65aa-42ec-a945-5fd21dec0538",
      "streetAddress1": "123 Main Ave",
      "streetAddress2": "Apartment 9000",
      "streetAddress3": "Montmârtre",
      "city": "Anytown",
      "state": "AL",
      "postalCode": "90210",
      "country": "USA",
      "eTag": "string"
    },
    "destinationAddress": {
      "id": "d2a1a09e-f3ab-415c-86ec-55f699842010",
      "streetAddress1": "123 Main Ave",
      "streetAddress2": "Apartment 9000",
      "streetAddress3": "Montmârtre",
      "city": "Anytown",
      "state": "AL",
      "postalCode": "90210",
      "country": "USA",
      "eTag": "string"
    },
    "agents": [
      {
        "firstName": "string",
        "lastName": "string",
        "email": "email@email.com",
        "phone": "916-039-3948",
        "agentType": "RELEASING_AGENT"
      }
    ]
  }
}
```

Run `prime-api-client`
```sh
prime-api-client create-mto-shipment --insecure --filename {filename of request payload}
```
Should return a successful response.

To see an error response, update the request payload. 
- Update the `moveTaskOrderID` to be a UUID that does not exist in our DB. Run the `prime-api-client` command from above, and you should see a 404 ClientError response.
- Remove the entire body from the request payload, run the same `prime-api-client` command as above, and you should see a 400 ClientError response.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2663) for this change

### CODE REVIEW

File: pkg/handlers/primeapi/mto_shipment.go [hunk 1]
```diff
@@ -52,7 +54,7 @@ func (h CreateMTOShipmentHandler) Handle(params mtoshipmentops.CreateMTOShipment
 		switch err.(type) {
 		case services.NotFoundError:
 			logger.Error("move task order not found", zap.Error(err))
-			return mtoshipmentops.NewCreateMTOShipmentNotFound()
+			return mtoshipmentops.NewCreateMTOShipmentNotFound().WithPayload(payloads.ClientError(handlers.NotFoundMessage, err.Error(), h.GetTraceID()))
```
Comment (sandy-wright) [inline pkg/handlers/primeapi/mto_shipment.go:57]: _Nevermind my previous comment, I can't read_

But also the errors in the `CreateMTOShipment` service item are not well formatted. In particular, there are a number of validation errors that could be raised in the transaction (like lines 74,84,93,96) that are just coming forward as generic errors. Should we make another ticket to fix those?

Comment (LeDeep) [reply pkg/handlers/primeapi/mto_shipment.go:57]: i should have checked the service. I assumed the service errors had already been updated. 🤦

Comment (LeDeep) [reply pkg/handlers/primeapi/mto_shipment.go:57]: added commit that updates the errors in the service, and i'll create a new ticket for resolving the issues with the IDs.

Comment (sandy-wright) [reply pkg/handlers/primeapi/mto_shipment.go:57]: the changes to the service object look good! Should probably add a case for `services.InvalidInputError` here though 😄

Comment (LeDeep) [reply pkg/handlers/primeapi/mto_shipment.go:57]: you don't like returning obscure 500 errors? 😆

**Pull request merged**